### PR TITLE
update(security): Ubuntu to alpine, update go pkgs

### PR DESCRIPTION
**Pull request by bryanpaget**
The Ubuntu image had some vulnerabilities and the easiest way to fix it was to switch to alpine. This results in a smaller image with fewer attack surfaces. Since this is a small image with a single application it makes sense to use a small base.

I've also updated some of the go packages, see the diff.

Jira: https://jirab.statcan.ca/browse/BTIS-388

Results:

- The new image works as expected. It has been tested on dev using a custom deployment. The new goofys container was able to mount the appropriate s3 shares.
- There are no CVEs in the image now according to the Trivy results.

```
pagetbr@l-pagetbr-3:~/workspace/projects/goofys$ trivy image goofys
2024-09-04T14:56:42-04:00	INFO	[vuln] Vulnerability scanning is enabled
2024-09-04T14:56:42-04:00	INFO	[secret] Secret scanning is enabled
2024-09-04T14:56:42-04:00	INFO	[secret] If your scanning is slow, please try '--scanners vuln' to disable secret scanning
2024-09-04T14:56:42-04:00	INFO	[secret] Please see also https://aquasecurity.github.io/trivy/v0.54/docs/scanner/secret#recommendation for faster secret detection
2024-09-04T14:56:42-04:00	INFO	Detected OS	family="alpine" version="3.20.2"
2024-09-04T14:56:42-04:00	INFO	[alpine] Detecting vulnerabilities...	os_version="3.20" repository="3.20" pkg_num=19
2024-09-04T14:56:42-04:00	INFO	Number of language-specific files	num=3
2024-09-04T14:56:42-04:00	INFO	[gobinary] Detecting vulnerabilities...

goofys (alpine 3.20.2)

Total: 0 (UNKNOWN: 0, LOW: 0, MEDIUM: 0, HIGH: 0, CRITICAL: 0)
```

### CODE REVIEW

**Comment by bryanpaget** [comment]
Since we ran into an issue with bash, I have added bash to the container in order to maximize compatibility since we were previously calling bash from the Ubuntu-based version of this goofys container.

**Comment by bryanpaget** [comment]
I tested this on dev by using a custom deployment provided by @mathis-marcotte, using this goofys image. I was able to shell into the running container and I was able to verify that the volume was being mounted as expected:

image

**Comment by bryanpaget** [comment]
Some new CVEs have popped up but adding apk update && apk upgrade to the Dockerfile ensures we are using the latest and most secure packages.

goofys (alpine 3.20.2)

Total: 2 (UNKNOWN: 0, LOW: 0, MEDIUM: 2, HIGH: 0, CRITICAL: 0)

┌────────────┬───────────────┬──────────┬────────┬───────────────────┬───────────────┬─────────────────────────────────────────────────────────┐
│  Library   │ Vulnerability │ Severity │ Status │ Installed Version │ Fixed Version │                          Title                          │
├────────────┼───────────────┼──────────┼────────┼───────────────────┼───────────────┼─────────────────────────────────────────────────────────┤
│ libcrypto3 │ https://github.com/advisories/GHSA-7m4m-pwhv-49c5 │ MEDIUM   │ fixed  │ 3.3.1-r3          │ 3.3.2-r0      │ Issue summary: Applications performing certificate name │
│            │               │          │        │                   │               │ checks (e.g., ...                                       │
│            │               │          │        │                   │               │ https://avd.aquasec.com/nvd/cve-2024-6119               │
├────────────┤               │          │        │                   │               │                                                         │
│ libssl3    │               │          │        │                   │               │                                                         │
│            │               │          │        │                   │               │                                                         │
│            │               │          │        │                   │               │                                                         │
└────────────┴───────────────┴──────────┴────────┴───────────────────┴───────────────┴─────────────────────────────────────────────────────────┘

**Comment by bryanpaget** [comment]
The CVEs have been resolved:

```
pagetbr@l-pagetbr-3:~/workspace/projects/goofys$ trivy image goofys
2024-09-04T14:56:42-04:00	INFO	[vuln] Vulnerability scanning is enabled
2024-09-04T14:56:42-04:00	INFO	[secret] Secret scanning is enabled
2024-09-04T14:56:42-04:00	INFO	[secret] If your scanning is slow, please try '--scanners vuln' to disable secret scanning
2024-09-04T14:56:42-04:00	INFO	[secret] Please see also https://aquasecurity.github.io/trivy/v0.54/docs/scanner/secret#recommendation for faster secret detection
2024-09-04T14:56:42-04:00	INFO	Detected OS	family="alpine" version="3.20.2"
2024-09-04T14:56:42-04:00	INFO	[alpine] Detecting vulnerabilities...	os_version="3.20" repository="3.20" pkg_num=19
2024-09-04T14:56:42-04:00	INFO	Number of language-specific files	num=3
2024-09-04T14:56:42-04:00	INFO	[gobinary] Detecting vulnerabilities...

goofys (alpine 3.20.2)

Total: 0 (UNKNOWN: 0, LOW: 0, MEDIUM: 0, HIGH: 0, CRITICAL: 0)
```

File: Dockerfile [hunk 1]
```diff
@@ -1,40 +1,51 @@
...
+RUN apk update && apk upgrade && apk --no-cache add git make gcc g++ libc-dev fuse-dev
 
 WORKDIR /goofys
 ADD . .
-# Builds the goofys app
+# Build the goofys app
 RUN make build
 
-FROM ubuntu:22.04
+FROM alpine:latest
```
Comment (Souheil-Yazji) [inline Dockerfile:21]: pin tag

File: Dockerfile [hunk 2]
```diff
@@ -1,40 +1,51 @@
-FROM golang:1.20.7 as fusermount3-proxy-builder
+FROM golang:1.23.0-alpine AS fusermount3-proxy-builder
```
Comment (bryanpaget) [inline Dockerfile]: revert to version 1.20.7

File: Dockerfile [hunk 3]
```diff
@@ -1,40 +1,51 @@
...
+RUN apk update && apk upgrade && apk --no-cache add make gcc g++ libc-dev fuse-dev
 
 WORKDIR /meta-fuse-csi-plugin
 ADD ./meta-fuse-csi-plugin .
-# Builds the meta-fuse-csi-plugin app
+# Build the fusermount3-proxy
 RUN make fusermount3-proxy BINDIR=/bin
 
-FROM golang:1.20.7 as goofys-builder
+FROM golang:1.23.0-alpine AS goofys-builder
```
Comment (bryanpaget) [inline Dockerfile]: revert to 1.20.7

**Review by Souheil-Yazji — CHANGES_REQUESTED** [review]

File: Dockerfile (hunk 1 above)
[the remark this answers is not in the document]
Comment (bryanpaget) [reply Dockerfile]: We are now using tag:3.20.3

**Review by bryanpaget — COMMENTED** [review]
I've updated the Dockerfile to address the issues we discussed during the tech-elab review.

**Review by mathis-marcotte — APPROVED** [review]
lgtm

**Review by Souheil-Yazji — APPROVED** [review]

**Pull request merged**